### PR TITLE
Refactor logger.New to accept FieldLogger

### PR DESCRIPTION
**Pull request by inancgumus**
While doing this I've also refactored the code a little bit. Let me know if you think the otherwise.

Closes: #823

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -31,21 +31,23 @@ func NewNullLogger() *Logger {
...
 	if logger == nil {
-		defLogger = true
-		logger = logrus.New()
+		ll = logrus.New()
+		ll.Warn("Logger", "no logger supplied, using default")
+	} else if l, ok := logger.(*logrus.Logger); ok {
+		ll = l
+	} else {
+		// panic so that we know when to upgrade. see issue #818.
+		panic("logger is not a logrus.Logger")
```
Comment (ankur22) [inline log/logger.go:44]: Could we work with the default (create a new logger like we do in the `nil` case) logger instead of panicking?

Comment (inancgumus) [reply log/logger.go:44]: Sure, but does it provide the same kind of security? I fear we might easily miss it.

Comment (inancgumus) [reply log/logger.go:44]: Ah, I see what you mean. You want to keep working with the `*logrus.Logger` no matter what. That's also an option, yes. WDYT? @na--?

Comment (ankur22) [reply log/logger.go:44]: Yeah, that makes sense. Could you try working with `ErrFatal` instead of `panic`?

Comment (inancgumus) [reply log/logger.go]: @ankur22 Please take a look: https://github.com/grafana/xk6-browser/pull/824/commits/3b2de1b241017bad1707289ca974a7264be3463b.

File: log/logger.go [hunk 2]
```diff
@@ -31,21 +31,21 @@ func NewNullLogger() *Logger {
...
-	var defLogger bool
+func New(logger logrus.FieldLogger, iterID string) *Logger {
+	ll := logrus.New()
+
 	if logger == nil {
-		defLogger = true
-		logger = logrus.New()
+		ll.Warn("Logger", "no logger supplied, using default")
+	} else if l, ok := logger.(*logrus.Logger); !ok {
+		ll.Warn("Logger", "invalid logger type, using default")
```
Comment (ankur22) [inline log/logger.go:40]: Isn't this a NPE?

Comment (inancgumus) [reply log/logger.go:40]: No, why?

**Review by ankur22 — APPROVED** [review]
LGTM

File: log/logger.go (hunk 2 above)
Comment (inancgumus) [reply log/logger.go:40]: Yep! The previous one was so :)

**Review by ka3de — APPROVED** [review]
LGTM.

**Pull request merged**